hazelcast/src/main/java/com/hazelcast/internal/util/ExceptionUtil.java

```diff
@@ -358,4 +359,22 @@ private static StackTraceElement[] getFixedStackTrace(Throwable throwable,
         System.arraycopy(localSideStackTrace, 1, newStackTrace, remoteStackTrace.length + 1, localSideStackTrace.length - 1);
         return newStackTrace;
     }
+
+    /**
+     * If there's any Throwable instance in the values, throw it.
+     */
+    @SafeVarargs
+    public static void rethrowFromCollection(Collection<?> values, Class<? extends Throwable> ... ignored) throws Throwable {
+        outerLoop:
```

Comment:
nit: would look better without the label usage:

```
Optional<Throwable> notIgnoredException = values.stream()
                .filter(Throwable.class::isInstance)
                .map(Throwable.class::cast)
                .filter(exception -> Arrays.stream(ignored)
                        .noneMatch(ignoredClass -> exception.getClass().isAssignableFrom(ignoredClass)))
                .findFirst();
if (notIgnoredException.isPresent()) {
    throw notIgnoredException.get();
}

Reply:
I agree it is more readable, but since it is the utility class it may be used in other places of our code in the future. This method is very straightforward, it will probably be never changed. For that reason I prefer version with better performance:;

```java
@State(Scope.Benchmark)
public class ExceptionUtil {
    private static final Class[] CLASSES = new Class[]{NullPointerException.class};
    private static final Collection COL = Collections.singletonList(new NullPointerException());

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    public void outerLoop(Blackhole blackhole) {
        try {
            rethrowFromCollectionOuter(COL, CLASSES);
        } catch (Throwable e) {
            blackhole.consume(e);
        }
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.NANOSECONDS)
    public void stream(Blackhole blackhole) {
        try {
            rethrowFromCollectionStream(COL, CLASSES);
        } catch (Throwable e) {
            blackhole.consume(e);
        }
    }

    @SafeVarargs
    public static void rethrowFromCollectionOuter(Collection<?> values, Class<? extends Throwable> ... ignored) throws Throwable {
        outerLoop:
        for (Object value : values) {
            if (value instanceof Throwable) {
                for (Class<? extends Throwable> ignoredClass : ignored) {
                    if (ignoredClass.isAssignableFrom(value.getClass())) {
                        continue outerLoop;
                    }
                }
                throw (Throwable) value;
            }
        }
    }

    @SafeVarargs
    public static void rethrowFromCollectionStream(Collection<?> values, Class<? extends Throwable> ... ignored) throws Throwable {
        Optional<Throwable> notIgnoredException = values.stream()
                .filter(Throwable.class::isInstance)
                .map(Throwable.class::cast)
                .filter(exception -> Arrays.stream(ignored)
                        .noneMatch(ignoredClass -> exception.getClass().isAssignableFrom(ignoredClass)))
                .findFirst();
        if (notIgnoredException.isPresent()) {
            throw notIgnoredException.get();
        }
    }
}
```

Results::
```
Benchmark                Mode  Cnt    Score   Error  Units
ExceptionUtil.outerLoop  avgt   25    1.578 ± 0.029  ns/op
ExceptionUtil.stream     avgt   25  187.085 ± 9.020  ns/op
```